iothub/device/src/InternalClient.cs

```diff
@@ -153,15 +153,6 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
 
             if (options != null && options.FileUploadTransportSettings != null)
             {
-                var fileUploadTransportSettings = options.FileUploadTransportSettings;
-                if (fileUploadTransportSettings.ClientCertificate == null)
-                {
-                    // Kind of strange to take file upload transport settings from the non-file upload transport settings,
-                    // but this is the established behavior from when upload to blob didn't have transport settings,
-                    // and it shouldn't be broken for existing users
```

Comment:
The existing APIs will never use the `_fileUploadHttpTransportHandler`, so this statement isn't valid.

Reply:
Good catch. When I added the new APIs, I thought I also made it so the obsolete APIs used the dedicated http transport instead of creating a new one each time with no transport settings configured. Looks like I didn't though, so this is a good fix